socorro/signature/rules.py

```diff
@@ -216,19 +216,23 @@ def normalize_frame(
     ):
         """Normalizes a single frame
 
+        Takes a frame and returns the most meaningful single-string representation of
+        the frame. Takes into account the following in this order of precedence
+
         Returns a structured conglomeration of the input parameters to serve as a
         signature. The parameter names of this function reflect the exact names of the
         fields from the JSON MDSW frame output. This allows this function to be invoked
         by passing a frame as ``**a_frame``.
 
         """
+        # If there's a function symbol, use that--it's the best
```

Comment:
I restructured this method to make it structurally like the precedence rules we want and less all-over-the-place if/then/else-wise which was hard to follow.